src/foundry/common/primitives/set.mjs.d.ts

```diff
@@ -0,0 +1,99 @@
+   * @returns Is the other set a subset of this one?
+   */
+  isSubset(other: Set<T>): boolean;
+
+  /**
+   * Convert a set to a JSON object by mapping its contents to an array
+   * @returns The set elements as an array.
+   */
+  toObject(): T[];
+
+  /**
+   * Test whether every element in this Set satisfies a certain test criterion.
+   * @see {@link Array#every}
+   * @param test - The test criterion to apply. Positional arguments are the value, the index of iteration, and the set being tested.
+   * @returns Does every element in the set satisfy the test criterion?
+   */
+  every(test: (value: T, index: number, set: Set<T>) => boolean): boolean;
+
+  /**
+   * Filter this set to create a subset of elements which satisfy a certain test criterion.
+   * @see {@link Array#filter}
+   * @param test - The test criterion to apply. Positional arguments are the value, the index of iteration, and the set being filtered.
+   * @returns A new Set containing only elements which satisfy the test criterion.
+   */
+  filter(test: (value: T, index: number, set: Set<T>) => boolean): Set<T>;
+
+  /**
+   * Find the first element in this set which satisfies a certain test criterion.
+   * @see {@link Array#find}
+   * @param test - The test criterion to apply. Positional arguments are the value, the index of iteration, and the set being searched.
+   * @returns The first element in the set which satisfies the test criterion, or undefined.
+   */
+  find(test: (value: T, index: number, set: Set<T>) => boolean): T | undefined;
+
+  /**
+   * Create a new Set where every element is modified by a provided transformation function.
+   * @see {@link Array#map}
+   * @param transform - The transformation function to apply.Positional arguments are the value, the index of iteration, and the set being transformed.
+   * @returns A new Set of equal size containing transformed elements.
+   */
+  map<V>(transform: (value: T, index: number, set: Set<T>) => V): Set<V>;
+
+  /**
+   * Create a new Set with elements that are filtered and transformed by a provided reducer function.
+   * @see {@link Array#reduce}
+   * @param reducer - A reducer function applied to each value. Positional arguments are the accumulator, the value, the index of iteration, and the set being reduced.
+   * @param accumulator - The initial value of the returned accumulator.
+   * @returns The final value of the accumulator.
+   */
+  reduce<V>(reducer: (accumulator: V, value: T, index: number, set: Set<T>) => V, accumulator: V): V;
+
+  /**
+   * Test whether any element in this Set satisfies a certain test criterion.
+   * @see {@link Array#some}
+   * @param test - The test criterion to apply. Positional arguments are the value, the index of iteration, and the set being tested.
+   * @returns Does any element in the set satisfy the test criterion?
+   */
+  some(test: (value: T, index: number, set: Set<T>) => boolean): boolean;
```

Comment:
```suggestion
  some(test: (value: T, index: number, set: Set<T>) => unknown): boolean;
```

Reply:
What is the point of this suggestion? It's explicitly documented as boolean upstream, unknown is overbroad and people should be explicit about returning a boolean anyway.

Reply:
Oh wait, it was tied to these https://github.com/League-of-Foundry-Developers/foundry-vtt-types/pull/2192#discussion_r946206283 https://github.com/League-of-Foundry-Developers/foundry-vtt-types/pull/2192#discussion_r946200670. I'm still going to stick to my comment about upstream since it asks for `boolean` not `unknown`, `any` or `*` for the test function return values.

Reply:
Hm, I suppose I’m fine with keeping `boolean`, but t we should add the type guard overload.